src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/ProcessLinkerXmlBase.cs

```diff
@@ -1,316 +1,510 @@
-﻿// Licensed to the .NET Foundation under one or more agreements.
+// Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
 using System;
 using System.Collections.Generic;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.Globalization;
+using System.IO;
+using System.Linq;
 using System.Reflection;
+using System.Reflection.Metadata;
 using System.Text;
+using System.Text.RegularExpressions;
 using System.Xml;
-
+using System.Xml.Linq;
+using System.Xml.XPath;
+using ILLink.Shared;
 using Internal.TypeSystem;
 
+#nullable enable
+
 namespace ILCompiler
 {
-    /// <summary>
-    /// Base class for readers of IL Linker XML file format.
-    /// </summary>
-    internal abstract class ProcessLinkerXmlBase
+    [Flags]
+    public enum AllowedAssemblies
     {
-        private readonly XmlReader _reader;
-        private readonly ModuleDesc _owningModule;
+        ContainingAssembly = 0x1,
+        AnyAssembly = 0x2 | ContainingAssembly,
+        AllAssemblies = 0x4 | AnyAssembly
+    }
+
+    public abstract class ProcessLinkerXmlBase
+    {
+        const string FullNameAttributeName = "fullname";
+        const string LinkerElementName = "linker";
+        const string TypeElementName = "type";
+        const string SignatureAttributeName = "signature";
+        const string NameAttributeName = "name";
+        const string FieldElementName = "field";
+        const string MethodElementName = "method";
+        const string EventElementName = "event";
+        const string PropertyElementName = "property";
+        const string AllAssembliesFullName = "*";
+        protected const string XmlNamespace = "";
+
+        // protected readonly string _xmlDocumentLocation;
```

Comment:
Could you uncomment this and populate it from the constructors? It would be good to have the API surface already enforce providing this value.

If it complains about this being never read, add a dummy `_ = _xmlDocumentLocation` next to one of the commented out throws.